File: data/src/main/java/com/crhonvas/data/mappers/user/ApiGeoToDomainMapper.java
package com.crhonvas.data.mappers.user;

import android.support.annotation.NonNull;

import com.crhonvas.data.remote.model.user.GeoApiResponse;
import com.crhonvas.domain.model.user.Geo;

import javax.inject.Inject;

//TODO: Inject scheduler provider inorder to do mappings in a specified background thread

/**
 * Mapper from {@link GeoApiResponse} to {@link Geo}
 */
public class ApiGeoToDomainMapper {

    @Inject
    public ApiGeoToDomainMapper() {}

    @NonNull
    public Geo map(@NonNull GeoApiResponse geoApiResponse) {
        if (geoApiResponse == null) {
            return new Geo.Builder().build();
        }
        return new Geo.Builder()
                .lat(geoApiResponse.getLat())
                .lng(geoApiResponse.getLng())
                .build();
    }
}

File: data/src/main/java/com/crhonvas/data/mappers/user/ApiGeoToDomainMapper.kt
package com.crhonvas.data.mappers.user

import com.crhonvas.data.remote.model.user.GeoApiResponse
import com.crhonvas.domain.model.user.Geo
import javax.inject.Inject

//TODO: Inject scheduler provider inorder to do mappings in a specified background thread
/**
 * Mapper from [GeoApiResponse] to [Geo]
 */
class ApiGeoToDomainMapper @Inject constructor() {
    fun map(geoApiResponse: GeoApiResponse): Geo {
        return if (geoApiResponse == null) {
            Geo.Builder().build()
        } else Geo.Builder()
            .lat(geoApiResponse.lat)
            .lng(geoApiResponse.lng)
            .build()
    }
}